services/explorer/db/sql/model.go

```diff
@@ -77,6 +77,50 @@ var (
 // PageSize is the amount of entries per page of events.
 var PageSize = 100
 
+// CCTPEvent stores a cctp event.
+type CCTPEvent struct {
+	// TxHash is the transaction hash of the event.
+	TxHash string `gorm:"column:tx_hash"`
+	// ContractAddress is the address of the contract that generated the event.
+	ContractAddress string `gorm:"column:contract_address"`
+	// BlockNumber is the timestamp.
+	BlockNumber uint64 `gorm:"column:block_number"`
+	// OriginChainID is the chain ID of the CCTP transfer.
+	OriginChainID *big.Int `gorm:"column:origin_chain_id;type:UInt256"`
+	// DestinationChainID is the chain ID of the CCTP transfer.
+	DestinationChainID *big.Int `gorm:"column:destination_chain_id;type:UInt256"`
+	// Sender is the sender of the CCTP transfer.
+	Sender sql.NullString `gorm:"column:sender"`
+	// Nonce is the nonce of the CCTP transfer.
+	Nonce sql.NullInt64 `gorm:"column:nonce"`
+	// BurnToken is the burn token of the CCTP transfer.
+	BurnToken sql.NullString `gorm:"column:burn_token"`
+	// MintToken is the mint token of the CCTP transfer.
+	MintToken sql.NullString `gorm:"column:mint_token"`
+	// SentAmount is the sent amount of the CCTP transfer.
+	SentAmount *big.Int `gorm:"column:sent_amount;type:UInt256"`
+	// SentAmountUSD is the sent amount of the CCTP transfer in USD terms.
+	SentAmountUSD *float64 `gorm:"column:sent_amount_usd;type:Float64"`
+	// ReceivedAmount is the received amount of the CCTP transfer.
+	ReceivedAmount *big.Int `gorm:"column:received_amount;type:UInt256"`
+	// RequestVersion is the request version of the CCTP transfer.
+	RequestVersion sql.NullInt32 `gorm:"column:request_version"`
+	// FormattedRequest is the formatted request of the CCTP transfer.
+	FormattedRequest []byte `gorm:"column:formatted_request"`
+	// RequestID is the request ID of the CCTP transfer.
+	RequestID [32]byte `gorm:"column:request_id"`
+	// Recipient is the recipient of the CCTP transfer.
+	Recipient sql.NullString `gorm:"column:recipient"`
+	// Fee is the fee of the CCTP transfer.
+	Fee *big.Int `gorm:"column:fee;type:UInt256"`
+	// FeeUSD is the fee of the CCTP transfer in USD terms.
+	FeeUSD *float64 `gorm:"column:fee_usd;type:Float64"`
+	// Token is the address of the received token.
+	Token sql.NullString `gorm:"column:token"`
```

Comment:
what will we be using this column for if we have burn and mint

also are we gonna use/cool with using fee as the difference between origin/destination?

Reply:
another idea is:
- have a non-pointer based column for chain id. put the chain where the event occurred in there
- have destination or origin populated depending on the event

Reply:
i think using fee is good too though

Reply:
Hmm yeah probably easiest to use fee as a switch for Sent/Fulfilled events.

I think the ideal way to do this would be to have `RequestID` be the primary key and handle OnConflict clauses to only update non-null fields if a record already exists for the given `RequestID`. Not sure where this logic would go though (also seems to break the pattern that you use in other event processing).

Another possibility would be to have separate models for Sent/Fulfilled events and leave it up to the db reader to join on `RequestID`, but again, depends on explorer access patterns and would require more boilerplate.